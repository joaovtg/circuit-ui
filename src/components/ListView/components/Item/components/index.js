export { default as Hover } from './Hover';
export { default as Wrapper } from './Wrapper';
