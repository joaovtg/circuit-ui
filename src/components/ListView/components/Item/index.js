export { Wrapper } from './components';
export { default } from './Item';
