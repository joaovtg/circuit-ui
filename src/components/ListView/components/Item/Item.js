import React from 'react';
import PropTypes from 'prop-types';

import { Hover, Wrapper } from './components';
import { sizes } from '../../../../styles/constants';

const { KILO, MEGA, GIGA } = sizes;

const Item = ({ children, ...props }) => (
  <Wrapper {...props}>
    {children}
    <Hover />
  </Wrapper>
);

Item.KILO = KILO;
Item.MEGA = MEGA;
Item.GIGA = GIGA;

Item.propTypes = {
  /**
   * When true, shows the item with selected styles
   */
  selected: PropTypes.bool,
  /**
   * A Circuit UI spacings size.
   */
  padding: PropTypes.oneOf([KILO, MEGA, GIGA]),
  /**
   * Content of the list item.
   */
  children: PropTypes.node.isRequired
};

Item.defaultProps = {
  padding: Item.GIGA,
  selected: false
};

/**
 * @component
 */
export default Item;
