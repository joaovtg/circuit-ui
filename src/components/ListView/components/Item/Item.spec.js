import React from 'react';

import Item from './Item';

describe('Item', () => {
  /**
   * Style tests.
   */
  it('should render with default styles', () => {
    const actual = create(<Item>List item</Item>);
    expect(actual).toMatchSnapshot();
  });

  it('should render with all paddings', () => {
    [Item.KILO, Item.MEGA, Item.GIGA].forEach(padding => {
      expect(
        create(<Item padding={padding}>List item</Item>)
      ).toMatchSnapshot();
    });
  });

  it('should render children', () => {
    const wrapper = shallow(
      <Item>
        <div data-selector="child">text node</div>
      </Item>
    );
    const actual = wrapper.find('[data-selector="child"]');

    expect(actual).toHaveLength(1);
    expect(actual.text()).toEqual('text node');
  });

  it('should render with selected styles', () => {
    const actual = render(<Item isSelected>List item</Item>);
    expect(actual).toMatchSnapshot();
  });
});
