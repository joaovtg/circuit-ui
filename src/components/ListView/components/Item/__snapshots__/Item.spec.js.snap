// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render with all paddings 1`] = `
.circuit-0 {
  display: none;
  box-shadow: 0px 0px 0px 2px #3388FF;
  border-radius: 4px;
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 1;
}

.circuit-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border-bottom: 1px solid #D8DDE1;
  padding: 12px;
}

.circuit-2:hover .circuit-1 {
  display: block;
}

.circuit-2:first-child {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.circuit-2:last-child {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

<div
  className="circuit-2 circuit-3"
  selected={false}
>
  List item
  <div
    className="circuit-0 circuit-1"
  />
</div>
`;

exports[`Item should render with all paddings 2`] = `
.circuit-0 {
  display: none;
  box-shadow: 0px 0px 0px 2px #3388FF;
  border-radius: 4px;
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 1;
}

.circuit-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border-bottom: 1px solid #D8DDE1;
  padding: 16px;
}

.circuit-2:hover .circuit-1 {
  display: block;
}

.circuit-2:first-child {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.circuit-2:last-child {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

<div
  className="circuit-2 circuit-3"
  selected={false}
>
  List item
  <div
    className="circuit-0 circuit-1"
  />
</div>
`;

exports[`Item should render with all paddings 3`] = `
.circuit-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border-bottom: 1px solid #D8DDE1;
  padding: 24px;
}

.circuit-2:hover .circuit-1 {
  display: block;
}

.circuit-2:first-child {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.circuit-2:last-child {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.circuit-0 {
  display: none;
  box-shadow: 0px 0px 0px 2px #3388FF;
  border-radius: 4px;
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 1;
}

<div
  className="circuit-2 circuit-3"
  selected={false}
>
  List item
  <div
    className="circuit-0 circuit-1"
  />
</div>
`;

exports[`Item should render with default styles 1`] = `
.circuit-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border-bottom: 1px solid #D8DDE1;
  padding: 24px;
}

.circuit-2:hover .circuit-1 {
  display: block;
}

.circuit-2:first-child {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.circuit-2:last-child {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.circuit-0 {
  display: none;
  box-shadow: 0px 0px 0px 2px #3388FF;
  border-radius: 4px;
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 1;
}

<div
  className="circuit-2 circuit-3"
  selected={false}
>
  List item
  <div
    className="circuit-0 circuit-1"
  />
</div>
`;

exports[`Item should render with selected styles 1`] = `
.circuit-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border-bottom: 1px solid #D8DDE1;
  padding: 24px;
}

.circuit-2:hover .circuit-1 {
  display: block;
}

.circuit-2:first-child {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.circuit-2:last-child {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.circuit-0 {
  display: none;
  box-shadow: 0px 0px 0px 2px #3388FF;
  border-radius: 4px;
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 1;
}

<div
  class="circuit-2 circuit-3"
>
  List item
  <div
    class="circuit-0 circuit-1"
  />
</div>
`;
